src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
+
+    this->disableBlinking = true;
+    this->textIdIndex = 24; // "Unclear which way things will go, cus Mutoh said he would call wife"
+    Message_BombersNotebookQueueEvent(play, BOMBERS_NOTEBOOK_PERSON_MAYOR_DOTOUR);
+
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_60_40)) {
+        this->textIdIndex = 26; // "Hard to figure out which way things will go"
+    }
+
+    this->actor.textId = sTextIds[this->textIdIndex];
+    this->eyeTexIndex = EN_DT_EYE_TEXTURE_LOOK_DOWN;
+    this->disableBlinking = true;
+    this->npcState = EN_DT_NPC_STATE_WAIT_FINAL_NIGHT_TALK;
+    this->actionFunc = EnDt_OfferFinalNightTalk;
+}
+
+void EnDt_OfferFinalNightTalk(EnDt* this, PlayState* play) {
+    EnDt_UpdateHeadRotate(this);
+
+    if (Actor_TalkOfferAccepted(&this->actor, &play->state)) {
+        EnDt_StartFinalNightTalk(this);
+        return;
+    }
+
+    // After completing Couple's Mask event and wearing Kafeis Mask
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80) && Player_GetMask(play) == PLAYER_MASK_KAFEIS_MASK) {
+        this->actor.textId = 0x2368; // "My wife hired you? Go ask the Curiosity Shop"
+    }
+
+    Actor_OfferTalk(&this->actor, play, 150.0f);
+}
+
+void EnDt_StartFinalNightTalk(EnDt* this) {
+    this->npcState = EN_DT_NPC_STATE_DONE_FINAL_NIGHT_TALK;
+    this->actionFunc = EnDt_TriggerFinalNightTalkEvent;
+}
 
-extern ColliderCylinderInit D_80BEB29C;
+void EnDt_TriggerFinalNightTalkEvent(EnDt* this, PlayState* play) {
+    EnDt_UpdateHeadRotate(this);
 
-extern UNK_TYPE D_0600112C;
+    if (Message_GetState(&play->msgCtx) == TEXT_STATE_EVENT && Message_ShouldAdvance(play)) {
+        Message_CloseTextbox(play);
+        if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_60_40)) {
+            this->textIdIndex = 25; // "The carnival is... on"
+            Message_ContinueTextbox(play, sTextIds[this->textIdIndex]);
+            SET_WEEKEVENTREG(WEEKEVENTREG_60_40);
+            return;
+        }
+        EnDt_SetupFinalNightState(this, play);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/EnDt_Init.s")
+void EnDt_Update(Actor* thisx, PlayState* play) {
+    EnDt* this = THIS;
+    ColliderCylinder* collider;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/EnDt_Destroy.s")
+    SkelAnime_Update(&this->skelAnime);
+    Actor_SetScale(&this->actor, 0.01f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9C74.s")
+    if (this->npcState != 4 && this->npcState != 5 && gSaveContext.save.day == 3 && gSaveContext.save.isNight) {
+        EnDt_SetupFinalNightState(this, play);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dt/func_80BE9CE8.s")
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80) &&
+        (gSaveContext.save.day != 3 || (gSaveContext.save.day == 3 && !gSaveContext.save.isNight))) {
+        Audio_PlaySequenceAtPos(SEQ_PLAYER_BGM_SUB, &gSfxDefaultPos, NA_BGM_MAYORS_OFFICE, 1000.0f);
+        Actor_PlaySfx(&this->actor, 0x205DU);
```

Comment:
If you run (note the trailing `.` is required)
```bash
./tools/sfxconvert.py src/overlays/actors/ovl_En_Dt/z_en_dt.c .
```
it should convert the 0x205D to the proper sfx.